src/pages/iv-user-defined/forms/UserTextForm.js

```diff
@@ -0,0 +1,208 @@
+      }
+    }
+    await postRequest({
+      extension: SystemRepository.Defaults.set,
+      record: JSON.stringify({ sysDefaults: dataToPost })
+    }).then(res => {
+      toast.success(platformLabels.Updated)
+      updateDefaults(dataToPost)
+    })
+  }
+
+  const handleSubmit = () => {
+    if (errored) {
+      return
+    }
+    formik.handleSubmit()
+  }
+
+  const [stagingDimCount, setStagingDimCount] = useState(formik.values.ivtUDTCount || null)
+
+  const handleDimCountChange = event => {
+    const value = event.target.value
+    setStagingDimCount(value)
+  }
+
+  function clearExcessFields(currentCount) {
+    for (let i = 1; i <= counter; i++) {
+      const dimKey = `ivtUDT${i}`
+      if (i > currentCount) {
+        formik.setFieldValue(dimKey, '', false)
+      }
+    }
+  }
+
+  const handleDimCountBlur = () => {
+    let value = parseInt(stagingDimCount, 10)
+    if (value > counter || !value || value < 1) {
+      setErrored(true)
+
+      return
+    }
+    setErrored(false)
+
+    setStagingDimCount(value)
+    formik.setFieldValue('ivtUDTCount', value)
+    clearExcessFields(value)
+  }
+
+  return (
+    <VertLayout>
+      <Grow>
+        <Grid container spacing={3} width={'50%'} sx={{ marginLeft: '0.5rem', marginTop: '0.5rem' }}>
+          <Grid item xs={12}>
+            <CustomNumberField
+              name='ivtUDTCount'
+              label={_labels.propertiesCount}
+              value={stagingDimCount === null ? formik.values.ivtUDTCount : stagingDimCount}
+              onChange={handleDimCountChange}
+              onBlur={handleDimCountBlur}
+              unClearable={true}
+              min={1}
+              max={20}
+              arrow={true}
+              maxLength={2}
+              decimalScale={0}
+              error={errored || (formik.touched.ivtUDTCount && Boolean(formik.errors.ivtUDTCount))}
+            />
+          </Grid>
+
+          <Grid item xs={12} md={6}>
+            <Grid container spacing={2}>
+              {Array.from({ length: 10 }).map((_, index) => (
+                <Grid item xs={12} key={index}>
+                  <CustomTextField
+                    key={index}
+                    name={`ivtUDT${index + 1}`}
+                    label={`${_labels.property} ${index + 1}`}
+                    value={formik.values[`ivtUDT${index + 1}`]}
+                    onClear={() => formik.setFieldValue(`ivtUDT${index + 1}`, '')}
+                    onChange={formik.handleChange}
+                    error={
+                      !errored && formik.values.ivtUDTCount > index && Boolean(formik.errors[`ivtUDT${index + 1}`])
+                    }
+                    inputProps={{
+                      readOnly: errored || formik.values.ivtUDTCount <= index || formik.values.ivtUDTCount === 'null'
+                    }}
+                  />
+                </Grid>
+              ))}
+            </Grid>
+          </Grid>
+
+          <Grid item xs={12} md={6}>
```

Comment:
Why did you utilize two CustomTextFields? Only CustomTextField with loop can be used.